prime-router/src/main/kotlin/cli/tests/Hl7Tests.kt

```diff
@@ -55,6 +57,54 @@ class Hl7Ingest : CoolTest() {
             passed = passed and examineLineageResults(reportId, receivers, itemCount)
         }
 
+        return passed
+    }
+}
+
+class BadHl7 : CoolTest() {
+    override val name = "badhl7"
+    override val description = "Submit badly formatted hl7 files - should get errors"
+    override val status = TestStatus.SMOKE
+    override fun run(environment: ReportStreamEnv, options: CoolTestOptions): Boolean {
+        val listBadCharacters = listOf(
+            "ÅÍÎÏ˝ÓÔ\uF8FFÒÚÆ☃",
+            "OBR|1|A241Z^MESA_ORDPLC||P2^Procedure 2^ERL_MESA|||||||||xxx||Radiology^^^^R|7101^ESTRADA^JAIME^P^^DR|||||||||||1^once^^20000701^^R|||WALK|Modality Test 241||||||||||A||\n",
+            "PID|||||Richards^Mary||19340428|F|||||||||||||||||||\nPID|||||||19700510105000|M|||||||||||||||||||\n", /* multiple patient information*/
+            "~!@#\\\$^&*()-_=+[]\\\\{}|;':,./<>?",
+            "❤️ \uD83D\uDC94 \uD83D\uDC8C \uD83D\uDC95 \uD83D\uDC9E \uD83D\uDC93",
+            "<a href=\"javascript\\x0A:javascript:alert(1)\" id=\"fuzzelement1\">test</a>",
+            "'; EXEC sp_MSForEachTable 'DROP TABLE ?'; --",
+            "1'000'000,00"
+        )
+        var passed = false
+        val sender = hl7Sender
+        listBadCharacters.forEachIndexed { i, badCharacters ->
+            ugly("Starting badcsv file Test $i: submitting with $badCharacters")
+            val reFile = FileUtilities.replaceText(
```

Comment:
Do not use this replaceText() as it.  I has a hard coded filename and folder.  Refactor that function first, so it does not use the same folder/file as other tests and we do not get a possible conflict.  You should also remove the temp file when you are done.  Better yet, read in the contents of the file and change the text in memory instead of a file.

Reply:
done